controller/src/beerocks/master/db/network_map.cpp

```diff
@@ -373,12 +417,26 @@ void network_map::send_bml_nodes_statistics_message_to_listeners(
             response =
                 message_com::create_vs_message<beerocks_message::cACTION_BML_STATS_UPDATE>(cmdu_tx);
 
-            beerocks_header                      = message_com::get_beerocks_header(cmdu_tx);
-            beerocks_header->actionhdr()->last() = 0;
-            num_of_stats_bulks                   = response->num_of_stats_bulks();
-            num_of_stats_bulks                   = 0;
-            get_next_node                        = 0;
-            size                                 = 0;
+            if (!response) {
+                LOG(ERROR) << "create_vs_message() has failed";
+                return -1;
+            }
+
+            auto beerocks_hdr = message_com::get_beerocks_header(cmdu_tx);
+            if (!beerocks_hdr) {
+                LOG(ERROR) << "get_beerocks_header() has failed";
+                return -1;
+            }
+            auto actionhdr = beerocks_hdr->actionhdr();
+            if (!actionhdr) {
+                LOG(ERROR) << "actionhdr() has failed";
+                return -1;
+            }
```

Comment:
Hm, this pattern is repeating a couple of times. And actually it is a bit silly: if create_vs_message doesn't return nullptr, then we're guaranteed to have a beerocks_hrd and an actionhdr...

Maybe we should just extend create_vs_message with an optional std::shared_ptr<beerocks_message::cACTION_HEADER> argument that gets updated with the action header for the cases that need it?

Reply:
I'll do it in next commit, greate idea